poh/src/poh_recorder.rs

```diff
@@ -314,7 +313,7 @@ impl PohRecorder {
         if let Some(WorkingBank { bank, start, .. }) = self.working_bank.take() {
             self.leader_bank_notifier.set_completed(bank.slot());
             let next_leader_slot = self.leader_schedule_cache.next_leader_slot(
-                &self.id,
+                bank.collector_id(),
```

Comment:
looks like this might have worked initially and then was changed from under us https://github.com/solana-labs/solana/pull/18182#discussion_r658931478

Reply:
Yup, I think the intention was to have `reset_poh_recorder` update the `id` in `PohRecorder` but that change was never implemented. I felt that it was better to remove the stored `id` entirely to prevent any future bugs.

Reply:
Just thinking out loud, there is no valid scenario here where `self.working_bank` is NOT one of our leader slots right ? That is, `bank.collector_id()` in this context should always be "us"

Reply:
It's only possible if you change your identity in the middle of your leader slots I think. But even if that happens, `clear_bank` being wrong isn't really a big deal from what I could tell

Reply:
> But even if that happens, clear_bank being wrong isn't really a big deal from what I could tell

Cool and yeah, we were doing it previously with stale ID so can't be too bad 😅